frontend/src/screens/admin/manage-learning-paths.tsx

```diff
@@ -0,0 +1,132 @@
+import { React } from '@common';
+import { useCreateLearningPath, useFetchLearningPaths, useUpdateLearningPath } from '../../models/learning-path';
+import { Button, Group, Loader, Stack, Textarea, TextInput, Title } from '@mantine/core';
+import { Main } from './grid';
+import { LearningPath } from '@api';
+import * as yup from 'yup';
+import { useForm, yupResolver } from '@mantine/form';
+
+export const ManageLearningPaths = () => {
+    const { data: learningPaths, isLoading } = useFetchLearningPaths()
+
+    if (isLoading) return <Loader />
+
+    return (
+        <Main className="container py-2">
+            <Title order={4}>Manage learning paths</Title>
+            <Stack>
+                <CreateLearningPath key='create-learning-path' />
+
+                {learningPaths?.map(learningPath => {
+                    return (
+                        <EditLearningPath key={learningPath.id} learningPath={learningPath} />
+                    )
+                })}
+            </Stack>
+        </Main>
+    )
+}
+
+const validationSchema = yup.object().shape({
+    label: yup.string().required(),
+    description: yup.string().required(),
+})
+
+const CreateLearningPath = () => {
+    const form = useForm<LearningPath>({
+        initialValues: {
+            label: '',
+            description: '',
+        },
+        validate: yupResolver(validationSchema),
+    })
+
+    type FormValues = typeof form.values;
+
+    const createLearningPath = useCreateLearningPath()
+
+    const handleSubmit = (values: FormValues) => {
+        createLearningPath.mutate({
+            learningPath: values,
+        }, {
+            onSuccess: () => {
+                form.reset()
+            },
+        })
+    }
+
+    return (
+        <>
+            <form onSubmit={form.onSubmit(handleSubmit)}>
+                <Stack>
+                    <TextInput
+                        withAsterisk
+                        label="Label"
+                        {...form.getInputProps('label')}
+                    />
+
+                    <Textarea
+                        withAsterisk
+                        label="Description"
+                        {...form.getInputProps('description')}
+                    />
+                </Stack>
+
+                <Group justify="flex-end" mt="md">
+                    <Button type="submit" disabled={!form.isValid()}>
+                        Create Learning Path
+                    </Button>
+                </Group>
+            </form>
+        </>
+    )
+}
```

Comment:
this seems like it's poorly typed and we're doing more work than needed.   Typically when I find myself in this situation it's better to let typescript do the inference.  this seems to work better for me with everything still being typed.

By moving the submit function inside the form.onSubmit, the values get typed due to form being typed.

```suggestion
const CreateLearningPath = () => {
    const form = useForm({
        initialValues: {
            label: '',
            description: '',
        } as LearningPath,
        validate: yupResolver(validationSchema),
    })

    const createLearningPath = useCreateLearningPath()

    const handleSubmit = form.onSubmit((values) => {
        createLearningPath.mutate({
            learningPath: values,
        }, {
            onSuccess: () => {
                form.reset()
            },
        })
    })

    return (
        <form onSubmit={handleSubmit}>
            <Stack>
                <TextInput
                    withAsterisk
                    label="Label"
                    {...form.getInputProps('label')}
                />

                <Textarea
                    withAsterisk
                    label="Description"
                    {...form.getInputProps('description')}
                />
            </Stack>

            <Group justify="flex-end" mt="md">
                <Button type="submit" disabled={!form.isValid()}>
                    Create Learning Path
                </Button>
            </Group>
        </form>
    )
}

```